mongo/database_test.go

```diff
@@ -83,16 +85,50 @@ func TestDatabase(t *testing.T) {
 		})
 	})
 	t.Run("replace topology error", func(t *testing.T) {
-		db := setupDb("foo")
+		t.Run("ErrClientDisconnected", func(t *testing.T) {
+			db := setupDb("foo")
+			err := db.RunCommand(bgCtx, bson.D{{"x", 1}}).Err()
+			assert.Equal(t, ErrClientDisconnected, err, "expected error %v, got %v", ErrClientDisconnected, err)
+
+			err = db.Drop(bgCtx)
+			assert.Equal(t, ErrClientDisconnected, err, "expected error %v, got %v", ErrClientDisconnected, err)
+
+			_, err = db.ListCollections(bgCtx, bson.D{})
+			assert.Equal(t, ErrClientDisconnected, err, "expected error %v, got %v", ErrClientDisconnected, err)
+		})
+		t.Run("TransientTransactionError", func(t *testing.T) {
+			var err error
+			client := setupClient(options.Client().ApplyURI("mongodb://nonexistent").SetServerSelectionTimeout(3 * time.Second))
+			err = client.Connect(bgCtx)
+			defer client.Disconnect(bgCtx)
+			assert.Nil(t, err, "expected nil, got %v", err)
 
-		err := db.RunCommand(bgCtx, bson.D{{"x", 1}}).Err()
-		assert.Equal(t, ErrClientDisconnected, err, "expected error %v, got %v", ErrClientDisconnected, err)
+			var sse topology.ServerSelectionError
+			var le LabeledError
 
-		err = db.Drop(bgCtx)
-		assert.Equal(t, ErrClientDisconnected, err, "expected error %v, got %v", ErrClientDisconnected, err)
+			// non-transaction
+			err = client.Ping(bgCtx, nil)
+			assert.NotNil(t, err, "expected error, got nil")
+			assert.True(t, errors.As(err, &sse), `expected error to be a "topology.ServerSelectionError"`)
+			if errors.As(err, &le) {
+				assert.False(t, le.HasErrorLabel("TransientTransactionError"), `expected error not to include the "TransientTransactionError" label`)
```

Comment:
```suggestion
				assert.False(t, le.HasErrorLabel(driver.TransientTransactionError), `expected error not to include the "TransientTransactionError" label`)
```

Reply:
I prefer to use the text explicitly over a variable defined by the driver code because it is the text that is specified in the specs. In this way, we can also cross-verify the variable value.